pkg/pillar/cmd/baseosmgr/baseosmgr.go

```diff
@@ -454,7 +456,7 @@ func handleGlobalConfigDelete(ctxArg interface{}, key string,
 	log.Infof("handleGlobalConfigDelete for %s\n", key)
 	debug, _ = agentlog.HandleGlobalConfig(ctx.subGlobalConfig, agentName,
 		debugOverride)
-	*ctx.globalConfig = types.GlobalConfigDefaults
+	*ctx.configItemValueMap = types.GetDefaultValueMap(ctx.configItemSpecMap)
```

Comment:
Don't need this one here.. we can never delete the "global" key. Just log it..

I see this is what is causing you to use the SpecMap in other agents..